x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -406,6 +406,34 @@ export class Authenticator {
     return AuthenticationResult.notHandled();
   }
 
+  /**
+   * Tries to reauthenticate request with the existing session.
+   * @param request Request instance.
+   */
+  async reauthenticate(request: KibanaRequest) {
+    assertRequest(request);
+
+    const existingSessionValue = await this.getSessionValue(request);
+    if (!existingSessionValue) {
+      this.logger.warn('It is not possible to extend session since it is no longer available.');
```

Comment:
I understand that a successful re-authentication always attempts to extend the session. However, if I was reading through the logs, I think I would find the log messages a bit confusing:

```
WARN: Re-authenticating request due to error: blah
WARN: It is not possible to extend session since it is no longer available.
```

Perhaps we could reword this to make it a bit clearer what's happening. WDYT about this?

```suggestion
      this.logger.warn('Session is no longer available and cannot be re-authenticated.');
```

Reply:
> Perhaps we could reword this to make it a bit clearer what's happening. WDYT about this?

Yeah, sounds better, thanks